### PR TITLE
Bug 1879047: bump prometheus-operator 0.42

### PR DESCRIPTION
**Pull request by s-urbaniak**
* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## 4.6
 
+- [#928](https://github.com/openshift/cluster-monitoring-operator/pull/928) Bump prometheus-operator 0.42:
+  - 0.42.0 changes: https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.42.0
+  - 0.41.1 changes: https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.41.1
+  - 0.41.0 changes: https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.41.0
```
Comment (s-urbaniak) [inline CHANGELOG.md:8]: is this sufficient for us? :thinking:

File: jsonnet/jsonnetfile.lock.json [hunk 2]
```diff
@@ -34,12 +34,12 @@
     {
       "source": {
         "git": {
-          "remote": "https://github.com/coreos/prometheus-operator.git",
+          "remote": "https://github.com/coreos/prometheus-operator",
```
Comment (simonpasquier) [inline jsonnet/jsonnetfile.lock.json:37]: shouldn't it be?

```suggestion
          "remote": "https://github.com/prometheus-operator/prometheus-operator",
```

Comment (s-urbaniak) [reply jsonnet/jsonnetfile.lock.json:37]: indeed, great catch, I focused on the version bump only :+1:

**Comment by s-urbaniak** [comment]
This e2e failure looks suspicious, investigating:
```
    alertmanager_test.go:273: request failed: Post "https://127.0.0.1:38151/api/v2/silences?namespace=test-kube-rbac-proxy": dial tcp 127.0.0.1:38151: connect: connection refused
```

File: CHANGELOG.md (hunk 1 above)
Comment (lilic) [reply CHANGELOG.md:8]: Yes even just bump to Prometheus operator 0.42 would be enough. 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## 4.6
 
+- [#928](https://github.com/openshift/cluster-monitoring-operator/pull/928) Bump prometheus-operator 0.42:
+  - 0.42.0 changes: https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.42.0
+  - 0.41.1 changes: https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.41.1
```
Comment (paulfantom) [inline CHANGELOG.md:7]: Those features are also included in 0.42, so including 0.41.1 changelog seems redundant

File: assets/prometheus-operator-user-workload/0probe-custom-resource-definition.yaml [hunk 4]
```diff
@@ -0,0 +1,212 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (lilic) [inline assets/prometheus-operator-user-workload/0probe-custom-resource-definition.yaml:1]: Why do we need the CRD in this directory?

File: jsonnet/jsonnetfile.lock.json (hunk 2 above)
Comment (s-urbaniak) [reply jsonnet/jsonnetfile.lock.json:37]: fixed (it was a long way)

File: assets/prometheus-operator-user-workload/0probe-custom-resource-definition.yaml (hunk 4 above)
Comment (s-urbaniak) [reply assets/prometheus-operator-user-workload/0probe-custom-resource-definition.yaml]: fixed (i think this was an intermediate state)

**Comment by s-urbaniak** [comment]
there is one thing missing: the cluster role binding to be allowed to operate on `Probe` custom resources.

**Comment by lilic** [comment]
@s-urbaniak should this be added in upstream or is this downstream patch we always do?

**Comment by s-urbaniak** [comment]
@lilic it is already present in upstream, but we refer to 0.40 cluster role bindings here. I am a bit unsure if this mixture of 0.40 and 0.42 assets is a good idea tbh :thinking: it introduces quite some intricate complexity.

**Comment by s-urbaniak** [comment]
changing strategy: we'll bump prometheus-operator 0.42 in `release-0.6` branch of kube-prometheus and bump the latter here instead of monkey patching jsonnet assets.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**